### PR TITLE
Ch/feature/refactor

### LINKED ISSUE #41: No Results Message
# Story 
As a user, I want to search for something and if there's no results, I want to see a message that there are no results. 

As sighted user, when I type "olay" in to the search bar and get back an error message that tells me nothing exists matching my search parameters.

### Concerns 
- [ ] conditional rendering only

### PR DESCRIPTION
**Pull request by conconartist**
- [x] Feature
- [x] Refactor
- [x] Bug Fix 
- [x] Clean Up

Refactor search and component and rendering for search if there are no matching results.  Cleaned up formatting and added hover and active states to images and links.  

Upon attempting to display "backup" images for broken image links, it was discovered that the images are valid links but the actual images are unavailable so error handling was created as an issue for a future iteration.

> Describe your code changes in detail for reviewers.

App.js was refactored in the return statement a well as in the SearchResults component.

View css files to observe changes made to active and hover states.

App.js and SearchResults.js were modified.

Run `npm start` in your terminal to view the site on localhost:3000 as well as start the server by entering nodeserver.js in another terminal window.  Enter a search term and you should see a message pop up if there are no results or you will see your results displayed.

Every thing seems to be working as intended by this point.  We still need to clean up tags for accessibility and refactor some of our larger methods but we should be on track! 

#### Checklist:
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] My code follows the style guidelines of this project
- [ ] I have used tests to prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my change

### CODE REVIEW

File: src/Components/App/App.js [hunk 1]
```diff
@@ -46,20 +46,20 @@ class App extends Component {
...
     })
-    const filterMakeupWithoutBrand = this.state.makeup.filter(item => item.brand)
-    const filteredMakeupBrand = filterMakeupWithoutBrand.filter(item => {
+    const filterWithBrand = this.state.makeup.filter(item => item.brand)
+    const filteredByBrand = filterWithBrand.filter(item => {
       return item.brand.toLowerCase().includes(userInput)
     })
-    const allFilteredMakeup = filteredMakeupTag.concat(filteredMakeupBrand)
-    const uniqueFilteredMakeup = [...new Set(allFilteredMakeup)]
+    const uniqueFilteredMakeup = [...new Set(filteredByTag.concat(filteredByBrand))]
```
Comment (gaj23) [inline src/Components/App/App.js:61]: This is a great addition to our app!

File: src/Components/App/App.js [hunk 2]
```diff
@@ -127,7 +135,16 @@ class App extends Component {
                 this.state.isSearching
                 ) {
                   return <Redirect to="/searchResults" />;
-                } else {
+              } else if(
```
Comment (gaj23) [inline src/Components/App/App.js:138]: I would like to create a helper function for this Route (is it possible to include helper functions on Router?).

File: src/Components/Form/Form.js [hunk 3]
```diff
@@ -11,7 +11,6 @@ const Form = ({ searchMakeup, handleClick}) => {
     event.preventDefault();
     searchMakeup(userInput);
     setUserInput("");
-    // handleClick()
```
Comment (gaj23) [inline src/Components/Form/Form.js:14]: Thank you for doing some clean up as well.

File: src/Components/LoadingImg/LoadingImg.css [hunk 4]
```diff
@@ -1,11 +1,11 @@
...
-    height: 20vh;
-    width: 20vh;
-    animation: pulse 4s ease infinite;
+  height: 20vh;
+  width: 20vh;
+  animation: pulse 4s ease infinite;
 }
 
 @keyframes pulse {
-    0% { transform: scale(1); }
```
Comment (gaj23) [inline src/Components/LoadingImg/LoadingImg.css:8]: Is there a setting on one of our text editors that causing this spacing issue?

File: src/Components/SearchResults/SearchResults.js [hunk 5]
```diff
@@ -1,31 +1,26 @@
 import React from 'react';
 import SearchResult from '../SearchResult/SearchResult';
 import './SearchResults.css';
-import {Link} from 'react-router-dom';
```
Comment (gaj23) [inline src/Components/SearchResults/SearchResults.js:4]: Thank you for this clean up.

**Review by gaj23 — APPROVED** [review]
This is a huge part and I'm looking forward to our last stretch with refactoring some final elements.

**Comment by gaj23** [comment]
I am a little surprised that error with Vercel hasn't cleared. Something to keep an eye on as we continue.

File: src/Components/LoadingImg/LoadingImg.css (hunk 4 above)
Comment (conconartist) [reply src/Components/LoadingImg/LoadingImg.css:8]: It might be a copy/paste thing, I think! @gaj23

File: src/Components/App/App.js (hunk 2 above)
Comment (conconartist) [reply src/Components/App/App.js:138]: We can look into it today! @gaj23

**Pull request merged**